common/persistence/serialization/task_serializers.go

```diff
@@ -14,101 +14,52 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-var _ TaskSerializer = (*taskSerializer)(nil)
-
-type (
-	TaskSerializer interface {
-		ReplicationTaskSerializer
-		SerializeTask(task tasks.Task) (*commonpb.DataBlob, error)
-		DeserializeTask(category tasks.Category, blob *commonpb.DataBlob) (tasks.Task, error)
-	}
-	ReplicationTaskSerializer interface {
-		SeralizeReplicationTask(task tasks.Task) (*persistencespb.ReplicationTaskInfo, error)
-		DeserializeReplicationTask(replicationTask *persistencespb.ReplicationTaskInfo) (tasks.Task, error)
-	}
-	taskSerializer struct {
-		serializer Serializer
-	}
-)
-
-func NewTaskSerializer(serializer Serializer) TaskSerializer {
-	return &taskSerializer{serializer: serializer}
-}
-
-func (s *taskSerializer) SerializeTask(
-	task tasks.Task,
-) (*commonpb.DataBlob, error) {
-	category := task.GetCategory()
-	switch category.ID() {
-	case tasks.CategoryIDTransfer:
-		return s.serializeTransferTask(task)
-	case tasks.CategoryIDTimer:
-		return s.serializeTimerTask(task)
-	case tasks.CategoryIDVisibility:
-		return s.serializeVisibilityTask(task)
-	case tasks.CategoryIDReplication:
-		return s.serializeReplicationTask(task)
-	case tasks.CategoryIDArchival:
-		return s.serializeArchivalTask(task)
-	case tasks.CategoryIDOutbound:
-		return s.serializeOutboundTask(task)
-	default:
-		return nil, serviceerror.NewInternalf("Unknown task category: %v", category)
-	}
+// TaskSerializer is an interface for serializing and deserializing tasks.
+// It is now part of the Encoder and Decoder interfaces.
+type TaskSerializer interface {
+	SerializeTask(task tasks.Task) (*commonpb.DataBlob, error)
+	DeserializeTask(category tasks.Category, blob *commonpb.DataBlob) (tasks.Task, error)
+	ReplicationTaskSerializer
 }
 
-func (s *taskSerializer) DeserializeTask(
-	category tasks.Category,
-	blob *commonpb.DataBlob,
-) (tasks.Task, error) {
-	switch category.ID() {
-	case tasks.CategoryIDTransfer:
-		return s.deserializeTransferTask(blob)
-	case tasks.CategoryIDTimer:
-		return s.deserializeTimerTask(blob)
-	case tasks.CategoryIDVisibility:
-		return s.deserializeVisibilityTask(blob)
-	case tasks.CategoryIDReplication:
-		return s.deserializeReplicationTask(blob)
-	case tasks.CategoryIDArchival:
-		return s.deserializeArchivalTask(blob)
-	case tasks.CategoryIDOutbound:
-		return s.deserializeOutboundTask(blob)
-	default:
-		return nil, serviceerror.NewInternalf("Unknown task category: %v", category)
-	}
+// ReplicationTaskSerializer is an interface for serializing and deserializing replication tasks.
+// It is now part of the Encoder and Decoder interfaces.
+type ReplicationTaskSerializer interface {
+	SerializeReplicationTask(task tasks.Task) (*persistencespb.ReplicationTaskInfo, error)
+	DeserializeReplicationTask(replicationTask *persistencespb.ReplicationTaskInfo) (tasks.Task, error)
 }
 
-func (s *taskSerializer) serializeTransferTask(
+func serializeTransferTask(
```

Comment:
Made all of these in this file functions, ie helpers. Renamed file to `task_serializers` (added `s`) to make that a bit clearer.